tools/unzip/unzip.xml

```diff
@@ -1,88 +1,125 @@
+    <description>Unzip a file</description>
+    <macros>
+        <token name="@TOOL_VERSION@">6.0</token>
+        <token name="@VERSION_SUFFIX@">0</token>
+        <token name="@PROFILE@">22.01</token>
+    </macros>
+    <requirements>
+        <requirement type="package" version="@TOOL_VERSION@">unzip</requirement>
+        <requirement type="package" version="1.34">tar</requirement>
+    </requirements>
+    <command> 
+    <![CDATA[
         mkdir ./out &&
         #if $input_file.is_of_type("zip"):
-          unzip
-          #if $extract_options['extract_all'] == 'False'
-            -p "$input_file" "$extract_options['pathtofile']" > out.data
-          #else
-            "$input_file" -d out/
-          #end if
+            unzip
+            #if $extract_options['extract_all'] == 'False'
+                -p "$input_file" "$extract_options['pathtofile']" > out.data
+            #else
+                "$input_file" -d out/
+            #end if
         #end if
         #if $input_file.is_of_type("tar"):
-          tar
-          #if $extract_options['extract_all'] == 'False'
-            xfO "$input_file" "$extract_options['pathtofile']" > out.data
-          #else
-            -xf "$input_file" -C ./out/
-          #end if
+            tar
+            #if $extract_options['extract_all'] == 'False'
+              xfO "$input_file" "$extract_options['pathtofile']" > out.data
+            #else
+              -xf "$input_file" -C ./out/
+            #end if
         #end if
-        ]]>
-  </command>
-  <inputs>
-    <param label="input_file" name="input_file" type="data" format="zip,tar"/>
-    <conditional name="extract_options">
-      <param label="Extract single file" name="extract_all" type="select">
-          <option selected="True" value="True">All files</option>
-          <option value="False">Single file</option>
-      </param>
-      <when value="True"></when>
-      <when value="False">
-        <param name="pathtofile" type="text" value="" label="Filepath">
-          <validator type="expression" message="No two dots (..) allowed">'..' not in value</validator>
-          <validator type="expression" message="No comma (,) allowed">',' not in value</validator>
-          <validator type="expression" message="No asterisk (*) allowed">'*' not in value</validator>
-          <validator type="expression" message="No leading slash (/) allowed">str(value)[0] != '/'</validator>
-        </param>
-      </when>
-    </conditional>
-  </inputs>
-  <outputs>
-        <data auto_format="true" name="unzipped_single" from_work_dir="out.data">
-          <filter>extract_options['extract_all'] == 'False'</filter>
+        &&
+
+        ## Flatten any potential directory structure, e.g.
+        ## out/sub1/subsub2/fileA.ext -> out/sub1_subsub2_fileA.ext
+        find out -type f | cut -d '/' -f 2- | grep '/' | xargs -I % sh -c "mv out/% out/\\$(echo % | tr / _)"
```

Comment:
do we need gnu-coreutil for find?

Reply:
oh and grep

Reply:
Unless we want to pin the version, gnu-coreutils is not needed. Not much change can happen here I guess?